packages/cfd/src/Containers/__tests__/investor-password-manager.spec.js

```diff
@@ -0,0 +1,141 @@
+import React from 'react';
+import { screen, render, waitFor, fireEvent, act } from '@testing-library/react';
+import InvestorPasswordManager from '../investor-password-manager';
+import { localize } from '@deriv/translations';
+
+jest.mock('@deriv/shared/src/services/ws-methods', () => ({
+    __esModule: true,
+    default: 'mockedDefaultExport',
+    WS: {
+        verifyEmail: jest.fn(() => Promise.resolve()),
+    },
+}));
+
+const validLengthMock = (value = '', options) =>
+    (options.min ? value.length >= options.min : true) && (options.max ? value.length <= options.max : true);
+
+const mock_errors = {
+    password: () => localize('Password should have lower and uppercase English letters with numbers.'),
+    repeated_chars_are_easy: () => localize('Repeats like "aaa" are easy to guess'),
+    repeated_patterns_are_easy: () => localize('Repeats like "abcabcabc" are only slightly harder to guess than "abc"'),
+    recent_years_are_easy: () => localize('Recent years are easy to guess'),
+};
+
+jest.mock('@deriv/shared/src/utils/validation/declarative-validation-rules.js', () => ({
+    getErrorMessages: jest.fn(() => ({
+        password_warnings: mock_errors,
+    })),
+    validLength: jest.fn(() => {
+        validLengthMock;
+    }),
+}));
+
+describe('<InvestorPasswordManager> ', () => {
+    const mock_props = {
+        error_message_investor: 'Forgot your password? Please reset your password.',
+        is_submit_success_investor: false,
+        multi_step_ref: { current: { nextStep: jest.fn() } },
+        onSubmit: jest.fn(),
+        setPasswordType: jest.fn(() => 'investor'),
+        toggleModal: jest.fn(),
+        validatePassword: jest.fn(),
+    };
+
+    it('should render investor password manager', async () => {
+        render(<InvestorPasswordManager {...mock_props} />);
+
+        expect(await screen.findByText(/new investor password/i)).toBeInTheDocument();
```

Comment:
This assertion is repeated in the next test case, so can be removed from here.

Reply:
Done